### PR TITLE
drivers: clock_control: stm32: fix typo and clean up unnecessary code

### PR DESCRIPTION
**Pull request by ndrs-pst**
- Use a code spell-checking tool to scan and correct spelling errors
- Removes unnecessary initialization of the local variable
where its value is guaranteed to be overwritten by subsequent operations.

### CODE REVIEW

**Review by erwango — DISMISSED** [review]

**Review by FRASTM — DISMISSED** [review]

File: drivers/clock_control/clock_stm32_ll_h7.c [hunk 1]
```diff
@@ -832,11 +832,11 @@ static int set_up_plls(void)
 
 int stm32_clock_control_init(const struct device *dev)
 {
-	int r = 0;
```
Comment (gautierg-st) [inline drivers/clock_control/clock_stm32_ll_h7.c:835]: r needs to be initialized in case CONFIG_CPU_CORTEX_M7 is not defined. This should fix CI failures

**Review by gautierg-st — CHANGES_REQUESTED** [review]

File: drivers/clock_control/clock_stm32_ll_h7.c (hunk 1 above)
Comment (ndrs-pst) [reply drivers/clock_control/clock_stm32_ll_h7.c:835]: @gautierg-st Thanks for your catch.
By the way, I decided to change the return path from `return r;` to `return 0;` instead.
To achieve unification among each change of `stm32_clock_control_init` on the other files.

**Comment by erwango** [comment]
@ndrs-pst Still a CI failure to fix.

**Comment by ndrs-pst** [comment]
> @ndrs-pst Still a CI failure to fix.

To pass CI, the changes should be made according to @gautierg-st's suggestion. 🫤

**Review by gautierg-st — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**